accounts-db/src/accounts_db.rs

```diff
@@ -2742,86 +2755,95 @@ impl AccountsDb {
     /// 1. one of the pubkeys in the store has account info to a store whose store count is not going to zero
     /// 2. a pubkey we were planning to remove is not removing all stores that contain the account
     fn calc_delete_dependencies(
-        purges: &HashMap<Pubkey, (SlotList<AccountInfo>, RefCount)>,
+        purges: &Vec<RwLock<HashMap<Pubkey, CleaningInfo>>>,
```

Comment:
we should drop the `RwLock` at some point when it is no longer necessary, probably:
let purges = purges.into_iter().map(|x| x.into_inner().unwrap()).collect::<Vec<_>>()
This simplifies fns like this one.

Reply:
we could also look for places where we are not executing in parallel where perhaps we should be. and maybe the rwlocks help us, but probably not.

Reply:
Maybe if we separate (key, value) into separate vectors as Haoran suggested we don't need RwLock at all.

Reply:
> we should drop the `RwLock` at some point when it is no longer necessary, probably: let purges = purges.into_iter().map(|x| x.into_inner().unwrap()).collect::<Vec<_>>() This simplifies fns like this one.

I think it only works like this

```
purges.into_iter().map(|x| {
    let y = x.read().unwrap();
    y.clone()
}).collect::<Vec<_>>()
```
because x is a shared reference and RwLock::into_inner cannot move out of a shared reference.

Reply:
this works and is reasonable (doesn't copy the hashmaps):
```
 let purges = candidates.into_iter().map(|x| std::mem::take(&mut *x.write().unwrap())).collect::<Vec<_>>();
 ```

I tried Box::into_inner(candidates), but it complained about not knowing the size at compile time. I don't have a lot of experience with `Box`. I'm only saying that at some point we can get rid of the `RwLock` in the candidates. They were convenient early but no downstream code uses them at the moment. It only adds complexity and the possibility of deadlocks and poisoned rwlocks...

This isn't necessary for this pr, I don't think. And, I think the rwlocks could enable better threading/higher performance in the future.

Reply:
fwiw, the separating keys/values is an interesting idea. I'd have to think more. I think what we have is fine for now.